compat/src/portals.js

```diff
@@ -1,82 +1,71 @@
-import { createElement, hydrate, render, __u as _unmount } from 'preact';
+import { createElement, render } from 'preact';
 
+/**
+ * @param {import('../../src/index').RenderableProps<{ context: any }>} props
+ */
 function ContextProvider(props) {
 	this.getChildContext = () => props.context;
 	return props.children;
 }
 
 /**
  * Portal component
+ * @this {import('./internal').Component}
  * @param {object | null | undefined} props
  *
- * TODO: this could use the "fake root node" trick from the partial hydration demo
+ * TODO: use createRoot() instead of fake root
  */
 function Portal(props) {
 	const _this = this;
 	let container = props._container;
-	let wrap = createElement(
-		ContextProvider,
-		{ context: _this.context },
-		props._vnode
-	);
 
 	_this.componentWillUnmount = function() {
-		let parent = _this._temp.parentNode;
-		if (parent) parent.removeChild(_this._temp);
-		_unmount(_this._wrap);
+		render(null, _this._temp);
+		_this._temp = null;
```

Comment:
Should we null out `_this.container` too? So we don't hold on to the DOM reference?